sdk/python/kfp/dsl/_pipeline_volume.py

```diff
@@ -69,9 +72,16 @@ def __init__(self,
                 init_volume["persistent_volume_claim"] = pvc_volume_source
         super().__init__(**init_volume, **kwargs)
         if not name_provided:
-            self.name = "pvolume-%s" % hashlib.sha256(
-                bytes(json.dumps(self.to_dict(), sort_keys=True), "utf-8")
-            ).hexdigest()
+            hash_value = hashlib.sha256(bytes(json.dumps(self.to_dict(),
+                                                         sort_keys=True),
+                                              "utf-8")).hexdigest()
+            name_prefix = "pvolume-"
+            # Name must be no more than 63 characters, so we will keep the last
```

Comment:
Is there any reason to keep the last chars, not the first chars?
It could just be:
```python
name = "pvolume-" + hash_value
name = name[0:63]

Reply:
I thought they would be _more unique_ but I can't seem to find any math supporting it.
So I've changed it the way you proposed, it's way more clean.